packages/integration-tests/utils/replayHelpers.ts

```diff
@@ -108,3 +120,60 @@ function getReplayPerformanceSpans(recordingEvents: RecordingEvent[]): Performan
     .filter(data => data.tag === 'performanceSpan')
     .map(data => data.payload) as PerformanceSpan[];
 }
+
+export function getFullRecordingSnapshots(replayRequest: Request): RecordingSnapshot[] {
+  const events = getDecompressedRecordingEvents(replayRequest) as RecordingEvent[];
+  return events.filter(event => event.type === 2).map(event => event.data as RecordingSnapshot);
+}
+
+function getDecompressedRecordingEvents(replayRequest: Request): RecordingEvent[] {
+  return replayEnvelopeRequestParser(replayRequest, 5) as RecordingEvent[];
+}
+
+/**
+ * Copy of the envelopeParser from ./helpers.ts, but with the ability
+ * to decompress zlib-compressed envelope payloads which we need to inspect for replay recordings.
+ * This parser can handle uncompressed as well as compressed replay recordings.
+ */
+const replayEnvelopeRequestParser = (request: Request | null, envelopeIndex = 2): Event => {
+  const envelope = replayEnvelopeParser(request);
+  return envelope[envelopeIndex] as Event;
+};
+
+const replayEnvelopeParser = (request: Request | null): unknown[] => {
+  // https://develop.sentry.dev/sdk/envelopes/
+  const envelopeBytes = request?.postDataBuffer() || '';
+
+  // first, we convert the bugger to string to split and go through the uncompressed lines
+  const envelopeString = envelopeBytes.toString();
+
+  const lines = envelopeString.split('\n').map(line => {
+    try {
+      return JSON.parse(line);
+    } catch (error) {
+      // If we fail to parse a line, we _might_ have found a compressed payload,
+      // so let's check if this is actually the case.
+      // This is quite hacky but we can't go through `line` because the prior operations
+      // seem to have altered its binary content. Hence, we take the raw envelope and
+      // look up the place where the zlib compression header(0x78 0x9c) starts
+      for (let i = 0; i < envelopeBytes.length; i++) {
+        if (envelopeBytes[i] === 0x78 && envelopeBytes[i + 1] === 0x9c) {
```

Comment:
Yikes, wish we had added some additional headers to determine the payload type. I wouldn't have expected split to change the binary content though.